src/main/java/com/arcao/geocaching4locus/search_nearest/task/DownloadNearestTask.java

```diff
@@ -161,6 +162,15 @@ protected Intent doInBackground(Void... params) throws Exception {
             wpt.setExtraOnDisplay(mContext.getPackageName(), UpdateActivity.class.getName(), UpdateActivity.PARAM_SIMPLE_CACHE_ID, wpt.gcData.getCacheID());
           }
 
+          // Disable logs when newest log is DNF, NM, NA type.
+          if(mPrefs.getBoolean(PrefConstants.DOWNLOADING_DISABLE_DNF_NM_NA_CACHES, false)) {
+            GeocachingLog cacheLog = wpt.gcData.logs.get(wpt.gcData.logs.size() - 1);
```

Comment:
I see a few problems with this line:
- this doesn't return newest log, but the oldest downloaded.
- IndexOutOfBoundsException can occurs, because user can have (set "Count of logs" to "0")[https://geocaching4locus.eu/manual/preferences#toc-downloading]